experimental/alp/alp/tuner.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
```

Comment:
Same comment re using the harness and metaprogramming IR from python in the future: this is really hacky (fine for a first commit).

This one should really be integrated closely with both the experts and the problem definitions.

There have been a lot of recent refactorings and QoL improvements by @ftynse and it will be much more future-proof to integrate on top of what he did.